cmd/check-spelling/kata-spell-check.sh

```diff
@@ -0,0 +1,346 @@
+	(echo "$count"; echo "$dict") > "$dict_file"
+
+	cp "${fragment_dir}/${rules_file_name}" "${rules_file}"
+}
+
+spell_check_file()
+{
+	local file="$1"
+
+	[ -z "$file" ] && die "need file to check"
+	[ -e "$file" ] || die "file does not exist: '$file'"
+
+	[ -e "$dict_file" ] || make_dictionary
+
+	info "Spell checking file '$file'"
+
+	# Determine the pandoc input format.
+	local pandoc_input_fmts
+	local pandoc_input_fmt
+
+	local pandoc_input_fmts=$(pandoc --list-input-formats 2>/dev/null || true)
+
+	if [ -z "$pandoc_input_fmts" ]
+	then
+		# We're using a very old version of pandoc that doesn't
+		# support listing its available input formats, so
+		# specify a default.
+		pandoc_input_fmt="markdown_github"
+	else
+		# Pandoc has multiple names for the gfm parser so find one of them
+		pandoc_input_fmt=$(echo "$pandoc_input_fmts" |\
+			grep -E "gfm|github" |\
+			head -1 || true)
+	fi
+
+	[ -z "$pandoc_input_fmt" ] && die "cannot find usable pandoc input format"
+
+	local stripped_doc
+
+	local pandoc_doc
+	local utf8_free_doc
+	local pre_hunspell_doc
+	local hunspell_results
+	local final_results
+
+	# First strip out all code blocks and convert all
+	# "quoted apostrophe's" ('\'') back into a single apostrophe.
+	stripped_doc=$("$strip_cmd" -i "$file" -)
+
+	# Next, convert the remainder it into plain text to remove the
+	# remaining markdown syntax.
+	#
+	# Before pandoc gets hold of it:
+	#
+	# - Replace pipes with spaces. This
+	#   fixes an issue with old versions of pandoc (Ubuntu 16.04)
+	#   which completely mangle tables into nonsense.
+	#
+	# - Remove empty reference links.
+	#
+	#   For example, this markdown
+	#
+	#       blah [`qemu-lite`][qemu-lite] blah.
+	#         :
+	#       [qemu-lite]: https://...
+	#
+	#   Gets converted into
+	#
+	#       blah [][qemu-lite] blah.
+	#         :
+	#       [qemu-lite]: https://...
+	#
+	#   And the empty set of square brackets confuses pandoc.
+	#
+	# After pandoc has processed the data, remove any remaining
+	# "inline links" in this format:
+	#
+	#     [link name](#link-address)
+	#
+	# This is strictly only required for old versions of pandoc.
+
+	pandoc_doc=$(echo "$stripped_doc" |\
+		tr '|' ' '  |\
+		sed 's/\[\]\[[^]]*\]//g' |\
+		pandoc -f "${pandoc_input_fmt}" -t plain - |\
+		sed 's/\[[^]]*\]([^\)]*)//g' || true)
+
+	# Convert the file into "pure ASCII" by removing all awkward
+	# Unicode characters that won't spell check.
+	#
+	# Necessary since pandoc is "clever" and will convert things like
+	# GitHub's colon emojis (such as ":smile:") into the actual utf8
+	# character where possible.
+	utf8_free_doc=$(echo "$pandoc_doc" | iconv -c -f utf-8 -t ascii)
+
+	# Next, perform the following simplifications:
+	#
+	# - Remove URLs.
+	# - Remove email addresses.
+	# - Replace most punctuation symbols with a space
+	#   (excluding a dash (aka hyphen!)
+	# - Carefully remove non-hyphen dashes.
+	# - Remove GitHub @userids.
+	pre_hunspell_doc=$(echo "$utf8_free_doc" |\
```

Comment:
my eyes! 👀

Reply:
Agreed! Mine are still in bandages. I'm taking touch typing to a whole new level :smile: